### PR TITLE
Minor bugfixes & cleanups

### PR DESCRIPTION
**Pull request by franz**

### CODE REVIEW

File: tests/workgroup/for_bug.cl [hunk 1]
```diff
@@ -43,7 +43,7 @@ test_kernel (void)
    barrier tail replication.
 */
 
-  for (i = 16; i >= 0; i--) {
+  for (i = 16; i > 0; i--) {
```
Comment (pjaaskel) [inline tests/workgroup/for_bug.cl]: Wouldn't it be more correct to just XFAIL this test for ARM32?

Comment (franz) [reply tests/workgroup/for_bug.cl]: I believe it wouldn't. From the description in the testcase:

> This produces a crash with 'repl' and an infinite loop with 'wiloops'.

The testcase is run with `POCL_WORK_GROUP_METHOD=workitemloops` and it didn't crash or go into an infinite loop, instead it produced wrong output. Also, i'm not changing the relation between the `for` and the `if` conditions in the test. So this looks like an entirely different bug triggered by something new in LLVM 7.0+ (it worked on ARM32 until 6.0).

Comment (pjaaskel) [reply tests/workgroup/for_bug.cl]: Fair enough.

**Comment by pjaaskel** [comment]
Can you add at least the minimum LLVM version bump to CHANGES?

**Comment by franz** [comment]
Added

**Pull request merged**